src/sqlfluff/core/parser/segments/base.py

```diff
@@ -1033,6 +1037,53 @@ def iter_patches(self, templated_str: str) -> Iterator[FixPatch]:
                 )
 
 
+class BracketedSegment(BaseSegment):
+    """A segment containing a bracketed expression."""
+
+    type = "bracketed"
+    additional_kwargs = ["start_bracket", "end_bracket"]
+
+    def __init__(
+        self,
+        *args,
+        start_bracket: Tuple[BaseSegment] = None,
```

Comment:
Why is it a tuple rather than just a `BaseSegment`?

Reply:
Good question. I did this initially so that `tuple` arithmetic was easy, but these should always be single segments 🤔. For someone reading the code raw, do you think `BaseSegment` would make more sense? If so I make change it to do that instead.

Reply:
I'm torn, because the syntax for creating a tuple "on the fly" is really obtuse and easy to overlook. Let's stick with the current code, but maybe add a clarifying comment that there'll be only one?